testdata/threescale-adapter-config.yaml

```diff
@@ -27,8 +23,13 @@ metadata:
 spec:
   template: authorization
   params:
+    subject:
+      user: request.query_params["user_key"] | request.headers["x-user-key"] | request.api_key | ""
+      properties:
+        app_id: request.query_params["app_id"] | request.headers["x-app-id"] | ""
+        app_key: request.query_params["app_key"] | request.headers["x-app-key"] | ""
```

Comment:
I couldn't get that to work. Thats what I thought initially. Cant seem to have nested maps. I will try out some other stuff in the mean time but it wont be hard to go from this to a working alternative.

Reply:
we could be interested in knowing whether these parameters are empty or they weren't specified, perhaps we can leave out the `""` and test for `nil`?

Reply:
(for a bit of context, there is in 3scale, under certain conditions, a semantic difference between specifying an empty `app_key` to the Service Management API and simply not specifying it at all)

Reply:
I don't think that's going to be achievable with the current client. The default string value in go is `""`.

 When we create the params, even if we don't set it it will become empty string. I think would need to go ahead and use pointers to strings in the client to get this behaviour. 

Happy to create an issue on it and fix, but my suggestion is leave this as is for now, wdyt?